File: c_files/raylib/basic_window_resizable.c
#include <assert.h>
#include <math.h>
#include <raylib.h>

static const int SCREEN_WIDTH  = 1280;
static const int SCREEN_HEIGHT = 720;

int main(void) {
    SetConfigFlags(FLAG_WINDOW_RESIZABLE);
    InitWindow(SCREEN_WIDTH, SCREEN_HEIGHT, "");
    SetTargetFPS(60);

    RenderTexture canvas = LoadRenderTexture(SCREEN_WIDTH, SCREEN_HEIGHT);
    assert(IsRenderTextureValid(canvas));

    Vector2 world_mouse;

    while (!WindowShouldClose()) {
        BeginTextureMode(canvas);
        ClearBackground(RAYWHITE);
        EndTextureMode();

        const float   scale  = fminf((float) GetScreenWidth() / (float) SCREEN_WIDTH,
                                  (float) GetScreenHeight() / (float) SCREEN_HEIGHT);
        const Vector2 offset = (Vector2){(GetScreenWidth() - (SCREEN_WIDTH * scale)) / 2.0f,
                                         (GetScreenHeight() - (SCREEN_HEIGHT * scale)) / 2.0f};

        const Rectangle source = (Rectangle){0, 0, SCREEN_WIDTH, -SCREEN_HEIGHT};
        const Rectangle dest
            = (Rectangle){offset.x, offset.y, SCREEN_WIDTH * scale, SCREEN_HEIGHT * scale};

        const Vector2 screen_mouse = GetMousePosition();
        world_mouse                = (Vector2){screen_mouse.x * scale, screen_mouse.y * scale};

        BeginDrawing();
        ClearBackground(BLACK);

        DrawTexturePro(canvas.texture, source, dest, (Vector2){0, 0}, 0.0f, WHITE);

        EndDrawing();
    }

    UnloadRenderTexture(canvas);

    CloseWindow();

    return 0;
}

File: python_files/raylib/basic_window_resizable.py
import pyray as p

SCREEN_WIDTH: int = 1280
SCREEN_HEIGHT: int = 720


def main() -> None:
    p.set_config_flags(p.ConfigFlags.FLAG_WINDOW_RESIZABLE)
    p.init_window(SCREEN_WIDTH, SCREEN_HEIGHT, "")
    p.set_target_fps(60)

    canvas: p.RenderTexture = p.load_render_texture(SCREEN_WIDTH, SCREEN_HEIGHT)
    assert p.is_render_texture_valid(canvas)

    world_mouse: p.Vector2

    while not p.window_should_close():
        p.begin_texture_mode(canvas)
        p.clear_background(p.RAYWHITE)
        p.end_texture_mode()

        scale: float = min(
            p.get_screen_width() / SCREEN_WIDTH, p.get_screen_height() / SCREEN_HEIGHT
        )

        offset: p.Vector2 = p.Vector2(
            (p.get_screen_width() - (SCREEN_WIDTH * scale)) / 2.0,
            (p.get_screen_height() - (SCREEN_HEIGHT * scale)) / 2.0,
        )

        source: p.Rectangle = p.Rectangle(0, 0, SCREEN_WIDTH, -SCREEN_HEIGHT)
        dest: p.Rectangle = p.Rectangle(
            offset.x, offset.y, source.width * scale, source.height * scale
        )

        screen_mouse: p.Vector2 = p.get_mouse_position()
        world_mouse = p.Vector2(screen_mouse.x * scale, screen_mouse.y * scale)

        p.begin_drawing()
        p.clear_background(p.BLACK)

        p.draw_texture_pro(canvas.texture, source, dest, p.Vector2(0, 0), 0.0, p.WHITE)

        p.end_drawing()

    p.unload_render_texture(canvas)

    p.close_window()


if __name__ == "__main__":
    main()
